projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-users-endpoint.php

```diff
@@ -181,14 +181,16 @@ public function callback( $path = '', $blog_id = 0 ) {
 		) : array();
 		$viewers = array_map( array( $this, 'get_author' ), $viewers );
 
-		// we restrict search field to name when include_viewers is true.
+		// When include_viewers is true, search by username or email.
 		if ( $include_viewers && ! empty( $args['search'] ) ) {
 			$viewers = array_filter(
 				$viewers,
 				function ( $viewer ) use ( $args ) {
+					// Convert to WP_User so expected fields are available.
```

Comment:
Would it make sense or be possible to try to make that conversion earlier in the process, like for example make the `get_author` method return an instance of `WP_User`?

Reply:
I'd be reluctant to do that because `get_author` is a method of the parent class and it's used all over the API. The shape of the `WP_User` is quite different from the current return value of `get_author`, so there'd have to be a lot of adjustments to account for that. It might make sense do do it if for most calls of `get_author` we're subsequently converting the result to a `WP_User` (which I don't believe is the case) but either way it should be addressed separately from this PR.